From 966342776690992ae2b09ecfad4e462c254cd395 Mon Sep 17 00:00:00 2001
From: ksooo <kai.sommerfeld@gmx.com>
Date: Sun, 27 Sep 2026 20:53:18 +0200
Subject: [PATCH] Return the PCM buffer size from
 projectm_pcm_get_max_samples()

projectm_pcm_get_max_samples() is documented to return the number of
samples stored per channel, but returned WaveformSamples (480) instead
of AudioBufferSamples (576). Applications that limit the samples they
pass per frame to this value leave the oldest part of the analyzed
buffer filled with stale data, which causes visible jumps in the
waveform.

Cherry-picked from upstream commit dd89dfba0 (PR #1032), without the test.
---
 src/libprojectM/ProjectMCWrapper.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/libprojectM/ProjectMCWrapper.cpp b/src/libprojectM/ProjectMCWrapper.cpp
index c383018a3..04bda086d 100644
--- a/src/libprojectM/ProjectMCWrapper.cpp
+++ b/src/libprojectM/ProjectMCWrapper.cpp
@@ -394,7 +394,7 @@ void projectm_set_window_size(projectm_handle instance, size_t width, size_t hei
 
 unsigned int projectm_pcm_get_max_samples()
 {
-    return libprojectM::Audio::WaveformSamples;
+    return libprojectM::Audio::AudioBufferSamples;
 }
 
 template<class BufferType>
